src/target/riscv/riscv-013.c

```diff
@@ -4702,27 +4702,22 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
 		if (result != ERROR_OK)
 			return result;
 
-		/* Read sbcs value.
-		 * At the same time, detect if DMI busy has occurred during the batch write. */
-		bool dmi_busy_encountered;
-		if (dm_op(target, &sbcs, &dmi_busy_encountered, DMI_OP_READ,
-				DM_SBCS, 0, false, true) != ERROR_OK)
-			return ERROR_FAIL;
+		bool dmi_busy_encountered = riscv_batch_was_batch_busy(batch);
```

Comment:
Let's suppose the batch_run() on line 4699 above ends with "busy" result. Can you please help me understand where the subsequent dtmcs.dmireset takes place? My understanding is that:

- No dmireset is performed as part of the batch.
- The subsequent dm_read() on line 4712 starts with DMI busy being still set. 
- The dm_op_timeout(), called from the dm_read(), is bound to fail once due to the busy. Then the dmi reset happens and the operation is retried within dm_op_timeout(), finally having a chance to succeed.

So fortunately, it looks like everything will work in the end, but we have broken the rule that whoever causes DMI busy should also clean up after itself (by dtmcs.dmireset).

To fix this, should we call `increase_dmi_busy_delay()` from `batch_run()`? I believe so.

Note that the other function `batch_run_timeout()` already contains such a call to `increase_dmi_busy_delay()`.

Reply:
Sure - by all means.